opensearchpy/connection/http_requests.py

```diff
@@ -207,7 +213,11 @@ def perform_request(  # type: ignore
             if isinstance(e, requests.Timeout):
                 raise ConnectionTimeout("TIMEOUT", str(e), e)
             raise ConnectionError("N/A", str(e), e)
-
+        
+        # Add the service time to the raw_data
+        if calculate_service_time:
+            raw_data = raw_data.rstrip()[:-1] + f', "__client": {{"Service_Time": "{duration}"}}' + '}'
```

Comment:
After a quick look, I think rather than attempting to modify the JSON string like this, it'd make more sense to just return the value from this function. ie. changing `return response.status_code, response.headers, raw_data` to `return response.status_code, response.headers, raw_data, duration` or creating some 'RequestMetrics` class that contains the duration (and potential additional future metrics) which is returned. Then in the transport class inserting that into the returned/deserialized object as `__metrics__` or something like that. 

Adding it to the return signature of the connection class gives a clear and uniform way of returning this information across all connection implementations.

Some one with more Python prowess will hopefully be able to give a better idea of what's actually idiomatic.

Reply:
+1 for this.